elements/lisk-p2p/src/peer/outbound.ts

```diff
@@ -168,7 +146,11 @@ export class OutboundPeer extends Peer {
 
 		outboundSocket.on('connect', async () => {
 			this.emit(EVENT_CONNECT_OUTBOUND, this._peerInfo);
-			await this._updatePeerOnConnect();
+			try {
+				await Promise.all([this.fetchStatus(), this.discoverPeers()]);
```

Comment:
Maybe we can make these calls separately and catch and emit specific events? let's say if a peer was able to call `fetchStatus` but failed at `fetchPeers` then it will throw `EVENT_FAILED_TO_COLLECT_PEER_DETAILS_ON_CONNECT ` which is not very clear that which RPC exactly failed.